app/assets/javascripts/admin/addon/routes/admin-permalinks-index.js

```diff
@@ -0,0 +1,8 @@
+import DiscourseRoute from "discourse/routes/discourse";
+
+export default class AdminPermalinksIndexRoute extends DiscourseRoute {
+  setupController(controller, model) {
+    super.setupController(...arguments);
+    controller.set("hasPermalinks", model.length > 0);
```

Comment:
does it have to be a prop on the controller or could the template check that length?

Reply:
The controller modifies the model object when filtering the list of permalinks, so we don't want the search field to be hidden when you type a search term and get nothing back (because that would make `model.length` 0.